### ISSUE TITLE
Data source refresh removes open popups

### ISSUE THREAD

**Issue by markstuart** [mediasuitenz/mappy#61]
This means that if you open a popup just before the refresh occurs, it closes again straight away.

**Comment by digitalsadhu** [comment]
This also occurs for layer relationships. i.e. if you do some things in a layer notify handler like hide a layer, when the layer refreshes, it shows again

**Comment by markstuart** [comment]
Yes, in general the data refreshing destroys the existing layer and creates a new one with the latest data.

We need a better way of refreshing the data. I think we are going to have to get into a *lot* of state tracking (unless we can think of a better way...)

**Comment by markstuart** [comment]
For Mappy users, at this stage the only decent strategy for keeping the map state consistent with user interactions is to only use `singlePoll` dataSource definitions, and expect the user to refresh to get the latest data (or refresh the page/map yourself in some way).
